src/topology/homotopy.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2021 Shing Tak Lam. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shing Tak Lam
+-/
+
+import topology.unit_interval
+import topology.algebra.ordered.proj_Icc
+
+/-!
+# Homotopy between functions
+
+In this file, we define a `homotopy` between two functions `f₀` and `f₁`.
+
+## Definitions
+
+* `homotopy f₀ f₁` is the type of homotopies between `f₀` and `f₁`
+* `homotopy.refl f₀` is the constant homotopy between `f₀` and `f₀`
+* `homotopy.symm f₀ f₁` is a `homotopy f₁ f₀` defined by reversing the homotopy
+* `homotopy.trans F G`, where `F : homotopy f₀ f₁`, `G : homotopy f₁ f₂` is a `homotopy f₀ f₂`
+  defined by putting the first homotopy on `[0, 1/2]` and the second on `[1/2, 1]`.
+-/
+
+noncomputable theory
+
+universes u v
+
+variables {X : Type u} {Y : Type v} [topological_space X] [topological_space Y]
+
+open_locale unit_interval
+
+/--
+The type of homotopies between two functions.
+-/
+
+-- There are functions for which there are no homotopies between them.
+@[nolint has_inhabited_instance]
+structure homotopy (f₀ f₁ : X → Y) :=
```

Comment:
Should this go in a namespace since we already have something called [`_root_.homotopy`](https://leanprover-community.github.io/mathlib_docs/algebra/homology/homotopy.html#homotopy)

Reply:
(This seems to be causing the linter failure)

Reply:
So the result of the discussion on Zulip seems to be that this should go in a namespace. Since this definition implies/requires that `f₀` and `f₁` are continuous, would it make sense to call this `continuous_map.homotopy`, and have `f₀ f₁ : C(X, Y)`?

Reply:
You can certainly try that. Makes sense to me.